x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/index.tsx

```diff
@@ -597,31 +625,32 @@ export const EndpointList = () => {
       );
     }
   }, [
-    canAccessFleet,
-    canReadEndpointList,
-    columns,
+    initialized,
+    listError,
     endpointsExist,
-    endpointPrivilegesLoading,
-    handleCreatePolicyClick,
-    handleDeployEndpointsClick,
-    handleSelectableOnChange,
+    canReadEndpointList,
+    canAccessFleet,
+    policyItemsLoading,
     hasPolicyData,
-    listError?.message,
-    loading,
     mutableListData,
-    onTableChange,
+    columns,
     paginationSetup,
-    policyItemsLoading,
-    policyItems,
-    selectedPolicyId,
+    onTableChange,
+    loading,
     setTableRowProps,
     sorting,
+    endpointPrivilegesLoading,
+    policyItems,
+    handleDeployEndpointsClick,
+    selectedPolicyId,
+    handleSelectableOnChange,
+    handleCreatePolicyClick,
   ]);
 
   return (
     <AdministrationListPage
       data-test-subj="endpointPage"
-      hideHeader={!endpointsExist}
+      hideHeader={!endpointsExist || !initialized || !!listError}
```

Comment:
Changed.

Reply:
I'd like to raise objection here - we should not memoize something for readability.
we can extract something as a constant for readability, and afterwards, but only if it's really needed, we can memoize it. (mostly for performance reasons: either because of a heavy calculation, or because the variable is reference based and re-creating it with every render would trigger child components receiving it as a prop to re-render as well)

here I think calling a function with a function object

Reply:
Yes absolutely. No need to memoize unless it is a complex computation. In this case it is sufficient to abstract it to a const.